src/components/Quiz/useQuizStore.ts

```
@@ -0,0 +1,395 @@
+  getCurrentAnswer: () => string | undefined;
+  getQuizTitle: () => string;
+  formatTime: (seconds: number) => string;
+  getUserAnswers: () => UserAnswer[];
+  getUnansweredQuestionsFromUserAnswers: () => number[];
+  getQuestionsGroupedBySubject: () => { [key: string]: Question[] };
+}
+
+export const useQuizStore = create<QuizState>()(
+  devtools(
+    (set, get) => {
+      let timerInterval: NodeJS.Timeout | null = null;
```

ESLint

[error] 'NodeJS' is not defined.

(no-undef)